src/components/environment-variables/environment-variables-list.js

```diff
@@ -0,0 +1,287 @@
+import React, { useEffect, useState } from 'react';
+import {
+  CircularProgress,
+  Icon,
+  Input,
+  Label,
+  Table,
+  Tooltip,
+  Typography,
+  Button,
+} from '@equinor/eds-core-react';
+import { edit, layers, restore_page, save } from '@equinor/eds-icons';
+import useSaveEnvVar from './use-save-env-var';
+import requestStates from '../../state/state-utils/request-states';
+import Alert from '../alert';
+import './style.css';
+
+const EnvironmentVariablesList = (props) => {
+  const {
+    appName,
+    envName,
+    componentName,
+    componentType,
+    includeRadixVars,
+    setContext,
+    envVars,
+    readonly,
+  } = props;
+  const [inEditMode, setInEditMode] = useState(false);
+  const updatableEnvVars = [];
+  const [saveState, saveFunc, resetState] = useSaveEnvVar({
+    appName,
+    envName,
+    componentName,
+    updatableEnvVars,
+  });
+  const [editableEnvVars, setEditableEnvVars] = useState([]);
+  const [hasNonRadixEnvVars, setHasNonRadixEnvVars] = useState(false);
+
+  function checkHasNonRadixEnvVars(edEnvVars) {
+    for (let i = 0; i < edEnvVars.length; i++) {
+      if (edEnvVars[i].origEnvVar.isRadixVariable) {
+        continue;
+      }
+      return true;
+    }
+    return false;
+  }
+
+  useEffect(() => {
+    if (inEditMode) {
+      return;
+    }
+    let edEnvVars = getEditableEnvVars(includeRadixVars, envVars);
+    setHasNonRadixEnvVars(checkHasNonRadixEnvVars(edEnvVars));
+    setEditableEnvVars(edEnvVars);
+  }, [includeRadixVars, inEditMode, envVars]);
+  const handleSetEditMode = () => {
+    setContext({ paused: true });
+    setInEditMode(true);
+  };
+  const handleSave = () => {
+    if (readonly) {
+      return;
+    }
+    const updatableEnvVars = getUpdatableEnvVars(editableEnvVars);
+    if (updatableEnvVars.length > 0) {
+      saveFunc({ appName, envName, componentName, updatableEnvVars });
+    }
+    setInEditMode(false);
+    setContext({ paused: false });
+  };
+  const handleReset = () => {
+    resetState();
+    setInEditMode(false);
+    setContext({ paused: false });
+  };
+  function getOriginalEnvVarToolTip(envVar) {
```

Comment:
Can we move all these functions outside of the **EnvironmentVariablesList** function, e.g. to line 17.
They make it a bit hard to follow the flow in **EnvironmentVariablesList**

Reply:
Moved all functions at the bottom of the class. Left only const-s.